python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -0,0 +1,1219 @@
+            Raised if either ``exposure`` or ``darkExposure`` do not
+            have their dark time defined.
+
+        See Also
+        --------
+        lsst.ip.isr.isrFunctions.darkCorrection
+        """
+        expScale = exposure.getInfo().getVisitInfo().getDarkTime()
+        if math.isnan(expScale):
+            raise RuntimeError("Exposure darktime is NAN.")
+        if darkExposure.getInfo().getVisitInfo() is not None \
+                and not math.isnan(darkExposure.getInfo().getVisitInfo().getDarkTime()):
+            darkScale = darkExposure.getInfo().getVisitInfo().getDarkTime()
+        else:
+            # DM-17444: darkExposure.getInfo.getVisitInfo() is None
+            #           so getDarkTime() does not exist.
+            self.log.warning("darkExposure.getInfo().getVisitInfo() does not exist. Using darkScale = 1.0.")
+            darkScale = 1.0
+
+        isrFunctions.darkCorrection(
+            maskedImage=exposure.getMaskedImage(),
+            darkMaskedImage=darkExposure.getMaskedImage(),
+            expScale=expScale,
+            darkScale=darkScale,
+            invert=invert,
+        )
+
+    def applyPostIsr(self, **kwargs):
+        # TODO add post-ISR (4th column of Eli's calibration boxes)
+        pass
+
+    @staticmethod
+    def extractCalibDate(calib):
+        """Extract common calibration metadata values that will be written to
+        output header.
+
+        Parameters
+        ----------
+        calib : `lsst.afw.image.Exposure` or `lsst.ip.isr.IsrCalib`
+            Calibration to pull date information from.
+
+        Returns
+        -------
+        dateString : `str`
+            Calibration creation date string to add to header.
+        """
+        if hasattr(calib, "getMetadata"):
+            if 'CALIB_CREATION_DATE' in calib.getMetadata():
+                return " ".join((calib.getMetadata().get("CALIB_CREATION_DATE", "Unknown"),
+                                 calib.getMetadata().get("CALIB_CREATION_TIME", "Unknown")))
+            else:
+                return " ".join((calib.getMetadata().get("CALIB_CREATE_DATE", "Unknown"),
+                                 calib.getMetadata().get("CALIB_CREATE_TIME", "Unknown")))
+        else:
+            return "Unknown Unknown"
+
+    def doLinearize(self, detector):
+        """Check if linearization is needed for the detector cameraGeom.
+
+        Checks config.doLinearize and the linearity type of the first
+        amplifier.
+
+        Parameters
+        ----------
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector to get linearity type from.
+
+        Returns
+        -------
+        doLinearize : `Bool`
+            If True, linearization should be performed.
+        """
+        return self.config.doLinearize and \
+            detector.getAmplifiers()[0].getLinearityType() != NullLinearityType
+
+    def run(self, ccdExposure, dnlLUT=None, bias=None, deferredChargeCalib=None, linearizer=None,
```

Comment:
This should all be keywords (this is not quite in the dev guide yet, but it's the way we're going); if you don't have a default you can do `def run(self, *, ccdExposure, dnlLUT=None ...)`.